cmd/cluster-agent/api/v1/language_detection.go

```diff
@@ -0,0 +1,274 @@
+	patchOwner(deploymentName string, namespace string, containerslanguages *languagedetection.ContainersLanguages) error
+	patchAllOwners(ownerlanguages *OwnersLanguages)
+}
+
+type languagePatcher struct {
+	k8sClient dynamic.Interface
+}
+
+// newLanguagePatcher initializes and returns a new patcher with a dynamic k8s client
+func newLanguagePatcher() (*languagePatcher, error) {
+	apiCtx, apiCancel := context.WithTimeout(context.Background(), maximumWaitForAPIServer)
+	defer apiCancel()
+	apiCl, err := apiserver.WaitForAPIClient(apiCtx)
+
+	if err != nil {
+		return nil, err
+	}
+
+	k8sClient := apiCl.DynamicCl
+	return &languagePatcher{
+		k8sClient: k8sClient,
+	}, nil
+}
+
+func (lp *languagePatcher) getContainersLanguagesFromPodDetail(podDetail *pbgo.PodLanguageDetails) *languagedetection.ContainersLanguages {
+	containerslanguages := languagedetection.NewContainersLanguages()
+
+	for _, containerLanguageDetails := range podDetail.ContainerDetails {
+		container := containerLanguageDetails.ContainerName
+		languages := containerLanguageDetails.Languages
+		for _, language := range languages {
+			containerslanguages.Add(container, language.Name)
+		}
+	}
+
+	// Handle Init Containers separately
+	for _, containerLanguageDetails := range podDetail.InitContainerDetails {
+		container := fmt.Sprintf("init.%s", containerLanguageDetails.ContainerName)
+		languages := containerLanguageDetails.Languages
+		for _, language := range languages {
+			containerslanguages.Add(container, language.Name)
+		}
+	}
+
+	return containerslanguages
+}
+
+// Gets the containers languages for every owner
+func (lp *languagePatcher) getOwnersLanguages(requestData *pbgo.ParentLanguageAnnotationRequest) *OwnersLanguages {
+	ownerslanguages := make(OwnersLanguages)
+	podDetails := requestData.PodDetails
+
+	// Generate annotations for each owner
+	for _, podDetail := range podDetails {
+		ns := podDetail.Namespace
+		ownerRef := podDetail.Ownerref
+		ownerinfo := *newOwnerInfo(ownerRef.Name, ns, ownerRef.Kind)
+		ownerslanguages[ownerinfo] = lp.getContainersLanguagesFromPodDetail(podDetail)
+	}
+
+	return &ownerslanguages
+}
+
+func (lp *languagePatcher) getUpdatedOwnerAnnotations(currentAnnotations map[string]string, containerslanguages *languagedetection.ContainersLanguages) (map[string]string, int) {
+
+	if currentAnnotations == nil {
+		currentAnnotations = make(map[string]string)
+	}
+
+	// Add the existing language annotations into containers languages object
+	existingContainersLanguages := languagedetection.NewContainersLanguages()
+	existingContainersLanguages.ParseAnnotations(currentAnnotations)
+
+	// Append the potentially new languages to the containers languages object
+	languagesBeforeUpdate := existingContainersLanguages.TotalLanguages()
+	for container, languageset := range containerslanguages.Languages {
+		existingContainersLanguages.Parse(container, fmt.Sprint(languageset))
```

Comment:
So we have a `languageSet` then we serialize it to string and we parse it again to add to another `languageSet`.